src/KKManager/Windows/Content/CardWindow.cs

```diff
@@ -873,10 +867,225 @@ private void openInExplorerToolStripMenuItem_Click(object sender, EventArgs e)
+                        File.Move(card.Location.FullName, candidate);
+                        moved++;
+                    }
+                    catch { failed++; }
+                }
+                MessageBox.Show($"Moved {moved} file(s){(failed > 0 ? ", some failed" : "")}.", "Move cards", MessageBoxButtons.OK, MessageBoxIcon.Information);
+                RefreshList();
+            }
+        }
+
+        private void RebuildDuplicateFilter()
+        {
+            var activeSelectors = _duplicateKeySelectors
+                .Where(kvp => kvp.Key.Checked)
+                .Select(kvp => kvp.Value)
+                .ToList();
+
+            if (activeSelectors.Count == 0)
+            {
+                listView.AdditionalFilter = null;
+                listView.BuildList(true);
+                return;
+            }
+
+            var src = _typedListView.Objects;
+            if (src == null || src.Count == 0)
+            {
+                listView.AdditionalFilter = null;
+                listView.BuildList(true);
+                return;
+            }
+
+            string NormalizeToString(object v)
+            {
+                if (v == null) return "";
+                if (v is string s) return s.Trim().ToLowerInvariant();
+                if (v is IFormattable f) return f.ToString(null, System.Globalization.CultureInfo.InvariantCulture);
+                return v.ToString();
+            }
+
+            string CompositeKey(Card c) =>
+                string.Join("\u001F", activeSelectors.Select(sel => NormalizeToString(sel(c))));
+
+            var dupSet = new HashSet<Card>(
+                src.GroupBy(CompositeKey)
+                   .Where(g => g.Count() > 1)
+                   .SelectMany(g => g)
+            );
+
+            listView.AdditionalFilter = new ModelFilter(o => dupSet.Contains((Card)o));
+            listView.BuildList(true);
+        }
+
+        private void clearDuplicateFilterToolStripMenuItem_Click(object sender, EventArgs e)
+        {
+            foreach (var kvp in _duplicateKeySelectors.Keys)
+                kvp.Checked = false;
+
+            listView.AdditionalFilter = null;
+            listView.BuildList(true);
+        }
+
+        private void duplicatesByNameToolStripMenuItem_Click(object sender, EventArgs e)
+        {
+            RebuildDuplicateFilter();
+        }
+
+        private void duplicatesByFileSizeToolStripMenuItem_Click(object sender, EventArgs e)
+        {
+            RebuildDuplicateFilter();
+        }
+
+        private void duplicatesByFilenameToolStripMenuItem_Click(object sender, EventArgs e)
+        {
+            RebuildDuplicateFilter();
+        }
+
+        private void duplicatesByUserIDToolStripMenuItem_Click(object sender, EventArgs e)
+        {
+            RebuildDuplicateFilter();
+        }
+
+        private void duplicatesByDataIDToolStripMenuItem_Click(object sender, EventArgs e)
+        {
+            RebuildDuplicateFilter();
+        }
```

Comment:
These could be combined into a single event handler, simply set all of the toggles to use the same method.